design/design-doc-foreground-deletion.md

```diff
@@ -0,0 +1,205 @@
+# Foreground Cascading Deletion Support
```

Comment:
Nit: We prefer to wrap docs at 80 chars - I'd appreciate it if you could do so here. There's a few plugins (e.g. "rewrap" for VS Code) that can do this automatically.

Reply:
Fixed - sorry about that.